function_maps.go

```diff
@@ -91,9 +93,35 @@ func setResourceNameAnnotation(name string) string {
 	return fmt.Sprintf("gotemplating.fn.crossplane.io/composition-resource-name: %s", name)
 }
 
-func initInclude(t *template.Template) func(string, interface{}) (string, error) {
+func initTpl(parent *template.Template, includedNames map[string]int) func(string, interface{}) (string, error) {
```

Comment:
for reference the helm function: https://github.com/helm/helm/blob/c724175b032a1171a7e476d0bb5e86151645b710/pkg/engine/engine.go#L148

Reply:
@phisco are you suggesting changes to the function?